src/components/RequireAuth.test.tsx

```diff
@@ -0,0 +1,78 @@
+import { screen, render } from "@testing-library/react";
+import { AppServices } from "../abstractions";
+import { AppServicesProvider } from "./AppServicesContext";
+import { AppSessionStateStatusContext } from "./AppSessionStateContext";
+import { RequireAuth } from "./RequireAuth";
+
+describe(RequireAuth.name, () => {
```

Comment:
@Cromeror could you review these tests?

In the two first ones, I simply pass `appServices` to `RequireAuth` as a property, because no other component was involved in this behavior.

But in the last one, `RequireAuth` is using `NavigateToExternalUrl`, so, I need to store the `appServices` in the context using `AppServicesProvider`.

Do you think that I should use the last approach in all the three tests?

Reply:
Not, everything is ok right now is. 🚀